test/e2e/checkinterval_test.go

```diff
@@ -0,0 +1,99 @@
+//go:build e2e
+// +build e2e
+
+package e2e
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+	"github.com/openshift/cluster-ingress-operator/pkg/operator/controller"
+	ingresscontroller "github.com/openshift/cluster-ingress-operator/pkg/operator/controller/ingress"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	defaultHealthCheckInterval = 5 * time.Second
+	deploymentTimeout          = 1 * time.Minute
+)
+
+// NOTE: This test will mutate the default ingresscontroller
```

Comment:
Better to avoid using the default ingresscontroller if it isn't essential to the test or necessary for other reasons.  Using a custom ingresscontroller reduces side effects, and in this case, using `newPrivateController` should be sufficient (no need for a load balancer).

Reply:
Done.